### ISSUE TITLE
Another nested lambda capture bug

### ISSUE THREAD

**Issue by opqdonut** [carp-lang/Carp#366]
```
(defn main []
  (let [val 2]
    (println*
     &(Array.aupdate [[val]] 0
                     &(fn [nums] (Array.filter &(fn [num] (= val @num))
                                               @nums))))))
```
```
./out//main.c:2915:20: error: use of undeclared identifier 'val'
    _31_env->val = val;
```

**Comment by eriksvedang** [comment]
(Note: this needs to use `endo-filter` now)

The bug persists though.

**Comment by Vertmo** [comment]
This bug (as well as the one in #805) have been fixed by #850. These issues should probably be closed to avoid confusion.

**Comment by scolsen** [comment]
Confirmed that it works now. Thanks again @Vertmo, awesome fix, this is a huge improvement!
